cmd/earthly/main.go

```diff
@@ -1316,12 +1346,19 @@ func (app *earthlyApp) configureSatellite(cc cloud.Client) error {
 	return nil
 }
 
-func (app *earthlyApp) isUsingSatellite() bool {
-	return len(app.cfg.Satellite.Name) > 0
+func (app *earthlyApp) isUsingSatellite(c *cli.Context) bool {
+	if app.noSatellite {
+		return false
+	}
+	if c.IsSet("buildkit-host") {
+		// buildkit-host takes precedence
+		return false
+	}
+	return len(app.cfg.Satellite.Name) > 0 || app.satelliteName != ""
 }
 
 func (app *earthlyApp) GetBuildkitClient(c *cli.Context, cc cloud.Client) (*client.Client, error) {
-	err := app.configureSatellite(cc)
+	err := app.configureSatellite(c, cc)
```

Comment:
Honestly, `c` and `cc` is unreadable for me :( But I may be in the minority

Reply:
i was going to suggest `ctx`, but it's not the same `context.Context` as one would expect, so i'm fine with `c` in this case.

Reply:
`cliCtx` and `cloudClient` :D

Reply:
Done.